File: benchmarks/complex-test/backend/app/models/stress_check.py
"""
ストレスチェック関連のPydanticモデル
"""
from pydantic import BaseModel
from typing import Dict, Optional
from datetime import date


class StressCheckAnswer(BaseModel):
    """ストレスチェック回答リクエスト"""
    answers: Dict[str, int]  # { "q1": 4, "q2": 2, ... }


class StressCheckQuestion(BaseModel):
    """ストレスチェック質問"""
    id: str
    text: str
    category: str


class StressCheckResult(BaseModel):
    """ストレスチェック結果"""
    id: str
    period: date
    total_score: int
    is_high_stress: bool
    job_stress_score: float
    stress_reaction_score: float
    support_score: float
    satisfaction_score: float


class StressCheckHistoryItem(BaseModel):
    """ストレスチェック履歴アイテム"""
    id: str
    period: date
    total_score: int
    is_high_stress: bool


class NonTakenUser(BaseModel):
    """未受検者情報"""
    id: str
    email: str
    name: str  # メールアドレスのローカル部分を使用
    last_check_date: Optional[date]  # 最終受検日


class NonTakenUsersResponse(BaseModel):
    """未受検者一覧レスポンス"""
    period: date  # 対象期間
    deadline: Optional[date]  # 受検期限
    users: list[NonTakenUser]
    total_count: int  # 全従業員数
    non_taken_count: int  # 未受検者数


class DraftAnswerRequest(BaseModel):
    """途中保存リクエスト"""
    answers: Dict[str, int]  # { "q1": 4, "q2": 2, ... }


class DraftAnswerResponse(BaseModel):
    """途中保存レスポンス"""
    answers: Dict[str, int]
    updated_at: Optional[str] = None


class MigrateDraftRequest(BaseModel):
    """localStorage移行リクエスト"""
    answers: Dict[str, int]

File: benchmarks/complex-test/frontend/lib/api/stress-check.ts
/**
 * ストレスチェックAPI
 */
import apiClient from './client';

export interface StressCheckQuestion {
  id: string;
  text: string;
  category: string;
}

export interface StressCheckAnswer {
  answers: Record<string, number>;
}

export interface StressCheckResult {
  id: string;
  period: string;
  total_score: number;
  is_high_stress: boolean;
  job_stress_score: number;
  stress_reaction_score: number;
  support_score: number;
  satisfaction_score: number;
}

export interface StressCheckHistoryItem {
  id: string;
  period: string;
  total_score: number;
  is_high_stress: boolean;
}

export interface DraftAnswerResponse {
  answers: Record<string, number>;
  updated_at?: string;
}

export const stressCheckApi = {
  getQuestions: async (): Promise<{ questions: StressCheckQuestion[]; already_taken?: boolean; message?: string }> => {
    const response = await apiClient.get<{ questions: StressCheckQuestion[]; already_taken?: boolean; message?: string }>('/api/v1/stress-check/questions');
    return response.data;
  },

  submit: async (data: StressCheckAnswer): Promise<StressCheckResult> => {
    const response = await apiClient.post<StressCheckResult>('/api/v1/stress-check/submit', data);
    return response.data;
  },

  getHistory: async (): Promise<StressCheckHistoryItem[]> => {
    const response = await apiClient.get<StressCheckHistoryItem[]>('/api/v1/stress-check/history');
    return response.data;
  },

  getResult: async (checkId: string): Promise<StressCheckResult> => {
    const response = await apiClient.get<StressCheckResult>(`/api/v1/stress-check/result/${checkId}`);
    return response.data;
  },

  getDraft: async (): Promise<DraftAnswerResponse> => {
    const response = await apiClient.get<DraftAnswerResponse>('/api/v1/stress-check/draft');
    return response.data;
  },

  saveDraft: async (answers: Record<string, number>): Promise<DraftAnswerResponse> => {
    const response = await apiClient.post<DraftAnswerResponse>('/api/v1/stress-check/draft', { answers });
    return response.data;
  },

  deleteDraft: async (): Promise<void> => {
    await apiClient.delete('/api/v1/stress-check/draft');
  },

  migrateDraft: async (answers: Record<string, number>): Promise<DraftAnswerResponse> => {
    const response = await apiClient.post<DraftAnswerResponse>('/api/v1/stress-check/draft/migrate', { answers });
    return response.data;
  },
};
